src/main-process/atom-application.js

```diff
@@ -803,7 +803,7 @@ module.exports = class AtomApplication extends EventEmitter {
 
     this.disposable.add(
       ipcHelpers.on(ipcMain, 'resolve-proxy', (event, requestId, url) => {
```

Comment:
```js
      ipcHelpers.on(ipcMain, 'resolve-proxy', async (event, requestId, url) => {
        const proxy = await event.sender.session.resolveProxy(url);
        if (!event.sender.isDestroyed()) {
          event.sender.send('did-resolve-proxy', requestId, proxy);
		}
      })
```

Reply:
Can you explain this suggested change some?

I see the `async` keyword added. And `proxy` is set equal to the result of the Electron API call/Promise.

Is this just a readability change?

Reply:
This is the proper way of writing async code. The only reason that Electron makes the code promisified is to prevent callback hell. The power of async-await allows making the code's structure like normal code.
Learn more about this in this article:

[Asynchronous JavaScript: From Callback Hell to Async and Await](https://blog.hellojs.org/asynchronous-javascript-from-callback-hell-to-async-and-await-9b9ceb63c8e8)

Reply:
I've read about `async`/`await` in a couple of places since you posted this suggestion. `async` `await` is still Promises-based. I personally find the `async` `await` in this code harder to reason about at a glance than `functionThatReturnsAPromise().then()`. I like how explicit and natural-language-like `.then()` is. The "control flow" is very clear. Just a personal preference...

I am also not a huge fan of having an additional line of code / an additional `const` declaration added to this:
`const proxy = await event.sender.session.resolveProxy(url);`

 I believe this is mostly down to style... moving away from callbacks is the significant thing. `async`/`await` are just a different way to deal with Promises. If Atom's maintainers prefer `async` `await`, or if a bunch of people come in and comment they prefer `async` `await`, I will change it.

This does sounds beneficial though:
> 3. Debugging is much simpler. Setting a breakpoint inside a .then block will not move to the next .then because it only steps through synchronous code. But you can step through await calls as if they were synchronous calls.

So I'll think about it...

Reply:
It's the first time I see that someone finds ["The Pyramid of Doom"](https://en.wikipedia.org/wiki/Pyramid_of_doom_(programming)) clearer than ["Structured concurrency"](https://en.wikipedia.org/wiki/Structured_concurrency).

Reply:
A part of my reluctance to do this is that it would be such a minor change, and I'm already comfortable with my PR as I wrote it. And that's what I ran tests on. I don't want to go and validate this again. So I don't think it is required for me to see that suggestion and immediately jump to do it that way. Nothing personal.

There is also a lot of `.then()` in Atom's code-base. Especially around the Electron API call handling.

I feel like a style change PR that went around and changed the Promise usage to consistently use `async`/`await` (maybe limited to the Electron API calls to keep it a manageable amount of work and for ease of review?) would be a fine way to propose such a change, but none of the other recent API usage PRs for the Electron 9 PR have been using `async`/`await`. It feels extra inconsistent to start now for one of them.

If your feedback were regarding security or performance implications, I would respond to them with more urgency. As it is, this seems like a subjective choice, and I would respectfully decline to do it unless there is a maintainer who feels it should happen. Or if more people can demonstrate a consensus toward `async`/`await`. Maybe we can post a thing about it in the Discord.

Reply:
Admittedly there is much more `await` than `.then()` in Atom's codebase.

<details><summary>Counting lines with '.then' in Atom's code (click to expand):</summary>

```console
% git grep '\.then' | wc -l
     195
```

</details>

**195** instances of `.then` in the code.

<details><summary>Counting lines with 'await' in Atom's code (click to expand):</summary>

```console
% git grep 'await' | wc -l 
    1604
```

</details>

**1604** instances of `await` in the code.

Reply:
I asked about this in the discord and someone agreed with `async`/`await`. So that's two against one. I just updated this branch with `async`/`await`.

I also find it much nicer-looking when run through the linter that restores space-based indentation rather than tabs. That may have thrown me off even more. The tabs do strain the eyes a bit IMO.